### ISSUE TITLE
`has every` arrow function parameter seems to be equal to everything

### ISSUE THREAD

**Issue by ericmorand** [twigphp/Twig#3923]
```twig
{{ (5 has every v => v == 4 and v == "foo") ? 'Every' : 'Not every' }} v is equal to both 4 and "foo"
```

Outputs:

```text
Every v is equal to both 4 and "foo"
```

https://twigfiddle.com/0qaa9i

How can every _v_ be equal to both 4 _and_ "foo"?

That _v_ is equal to _one of them_ would be legit, but how can it be equal to both of them_? 4 is not equal to "foo" - either in PHP or Twig, so what is happening here?

Is this a bug? This is very hard to tell since there doesn't seem to be any documentation about the behavior of this operator outside of arrays.

**Comment by GromNaN** [comment]
You may have a PHP warning telling "Warning: foreach() argument must be of type array|object, int given". Since the `int` can't be iterated, the default value is returned, which is `true`.

https://github.com/twigphp/Twig/blob/aeeec9a5e907a79e50a6bb78979154599401726e/src/Extension/CoreExtension.php#L1731-L1742

**Comment by ericmorand** [comment]
I'm questioning the language specification itself, not the implementation.

The implementation was written the way it is written to honor the specification. Would the specification have established that "_it is not possible to use the `has every` operator on anything else than an iterable_", a runtime error would have been thrown by the implementation.

But since the implementation doesn't throw an error - and thus allows non-iterable values to be operand of the operator, it means that the specification says that it is allowed. That's why my question is about the rational of this specification and especially the one that drives the assertion "_in the body of the arrow function of the `has every` operator, the parameters are equal to everthing_".

If this is what the specification says, then the implementation is fine. But I think it needs to be explained in the documentation because it makes this operator unpredictable from a consumer point of view.

And from the point of view of the teams that implement the language and that have no way to know if there implementation is correct or not.

**Comment by ericmorand** [comment]
The `has every` operator seems to be all over the place:

```
{{ 5 has every "whatever" }}
{{ 5 has every true }}
{{ 5 has every null }}
{{ 5 has every false }}
{{ 5 has every (5 == 3) }}
```

Outputs:

```txt
1
1
1
1
1
```

What is the exact accepted syntax of this operator? And what is its purpose? Why does it accept basically anything as second operand?

**Comment by GromNaN** [comment]
If it's not covered by a test, it's just an edge case, not a specific feature. There isn't type hint on the argument because `iterable` type did not exist when Twig was created, and it was not added after.
The PHP warning is a clear indicator that you are not calling the function correctly. But I'm sure we can improve it.

What do your think should be the correct behavior? Throw a Type error, return true or false, or call the callback with the single value?

**Comment by fabpot** [comment]
You're right @ericmorand, it should only support `iterable`s. Fixed in 3523d34b50c97bf28f924ebef684aea4052a94fe.
